### DISCUSSION TITLE
Include additional custom items in `#quarto-sidebar > #toc-actions`

### DISCUSSION THREAD

**Discussion by kylebutts** [quarto-dev/quarto-cli#8276] [Q&A]
### Description

Hi there,

I'm new to writing quarto extensions, but I'm wondering if it's possible to add additional items to the toc actions section:
![TOC Actions Example](https://github.com/quarto-dev/quarto-cli/assets/19961439/ad095f4b-b768-4eb5-950b-fe8c4a6d6271)

I have some ideas for features on a quarto project, and would want an easy way to put a toggle button into the sidebar. Currently, I have a script that creates an html element and then when the page loads moves it to the sidebar, but it would be nice to streamline this.

For a concrete example, we have this online textbook with options for three languages: https://mixtape.scunning.com/03-directed_acyclical_graphs. I would love to have a button on the side that lets you toggle on one of the languages to be the default for the tabs.

**Comment by mcanouil** [comment]
There is no easy way as this is the Quarto envelope which is not currently accessible outside of Quarto.
But for your use case there are already several examples/discussions you can search in discussions/issues, for example:
- https://github.com/quarto-dev/quarto-cli/discussions/6319

**Comment by nicucalcea** [comment]
I have a slightly different use case. I'd like to add an option on some of my pages to open a notebook in Google Colab, so that readers can quickly open and run the code themselves.

In the absence of a more elegant solution, I added a post-render script that adds the link in once the website's been rendered. I've [got the script here](https://github.com/nicucalcea/ddj-wiki/blob/main/scripts/add_colab_link.py) and [an example of it on the website here](https://ddj.nicu.md/ai/python-classification-rag.html).

**Reply by mcanouil** [reply]
See https://quarto.org/docs/output-formats/html-basics.html#code-links-and-other-links which might be simpler.

**Reply by nicucalcea** [reply]
That is indeed better, thanks!
